department-of-veterans-affairs/vets-website
# MHV-56870 Address Refresh And Reload Issues In Medical Records List Views

## Summary

Updated `VitalDetails.jsx` and `RecordList.js` to ensure that focus goes to the H1 element instead of the records count. Focus transfers to pagination only when a user selects a pagination element.

## Related issue(s)

[MHV-56870](https://jira.devops.va.gov/browse/MHV-56870) - Address Refresh and Reload Issues in Medical Records List Views

Address Refresh and Reload Issues in Medical Records List Views

(Aria wide and focus state issue on browser refresh -everywhere where the page is getting reloaded, new tabs,.  Focus should go to the H1 instead of the records count.)

Currently, there are three instances of reload and refresh issues within the medical records list views. These issues likely stem from the same underlying problem, and fixing one instance is anticipated to resolve the others. The problems manifest in the following scenarios:

Browser Refresh:

When a user refreshes the browser, the focus state shifts to "Showing 1 to 10 of 118 records from newest to oldest." (no matter which domain they are on)

Browser Refresh followed by Side Navigation Click:

If a user refreshes the browser and subsequently clicks any links in the side navigation, the focus state again defaults to "Showing 1 to 10 of 118 records from newest to oldest."

New Tab Opening:

Opening a link in a new tab similarly results in the focus state being set to "Showing 1 to 10 of 118 records from newest to oldest." (no matter which domain they interact with)

User Story: As a Medical Records User, I want to   * , so that I can    

GIVEN

WHEN

THEN

Feature Flag Y/N ? N

DataDog/Analytics Y/N ? N

Manual Testing Y/N ? Y

Automated Testing Y/N ? N

Accessibility Testing Y/N ? Y

UCD Validation Y/N N

Acceptance Criteria:

AC1  Refresh and Reload Issues in Medical Records List Views have been addressed 

AC2 All 3 scenarios have been reviewed and tested

AC3
AC4
AC5

Links to UCD:

Figma Link:

User Flow:

Mobile:

Desktop:

Other Design Notes:

Architectural Notes: Always check for the latest designs in Figma

## Testing done

Conducted UI testing to ensure the expected behavior. I updated existing unit test for new components.

## Screenshots

![Screenshot 2024-05-31 at 12 37 20 PM (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/4b0de54d-a6a0-41eb-aca2-ae82dbd39731)
![image](https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/850f34fb-b78d-4491-945d-27188111e5cc)
![Screenshot 2024-05-31 at 11 13 29 AM (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/36acb775-2ef3-4b78-adfd-9c8d8b452e65)
![image](https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/3ed0737b-8316-49a4-9987-ace0e69d23c7)

## What areas of the site does it impact?

The modification affects VitalDetails and  RecordList of the Medical Records pages.

## Acceptance criteria
focus goes to the H1 element instead of the records count.

### Quality Assurance & Testing


### Authentication
